api/tools/src/main/java/org/pmiops/workbench/tools/BackfillCanCompute.java

```diff
@@ -0,0 +1,169 @@
+          .desc("If specified, the tool runs in dry run mode; no modifications are made")
+          .build();
+  private static Options options =
+      new Options().addOption(fcBaseUrlOpt).addOption(billingProjectPrefixOpt).addOption(dryRunOpt);
+
+  private static final String[] FC_SCOPES =
+      new String[] {
+        "https://www.googleapis.com/auth/userinfo.profile",
+        "https://www.googleapis.com/auth/userinfo.email"
+      };
+  private static final Logger log = Logger.getLogger(BackfillCanCompute.class.getName());
+  private static final Set<String> CAN_COMPUTE_ROLES =
+      ImmutableSet.of("PROJECT_OWNER", "OWNER", "WRITER");
+
+  private static WorkspacesApi newWorkspacesApi(String apiUrl) throws IOException {
+    ApiClient apiClient = new ApiClient();
+    apiClient.setBasePath(apiUrl);
+    GoogleCredential credential =
+        GoogleCredential.getApplicationDefault().createScoped(Arrays.asList(FC_SCOPES));
+    credential.refreshToken();
+    apiClient.setAccessToken(credential.getAccessToken());
+    WorkspacesApi api = new WorkspacesApi();
+    api.setApiClient(apiClient);
+    return api;
+  }
+
+  private static void dryLog(boolean dryRun, String msg) {
+    String prefix = "";
+    if (dryRun) {
+      prefix = "[DRY RUN] Would have... ";
+    }
+    log.info(prefix + msg);
+  }
+
+  /**
+   * Swagger Java codegen does not handle the WorkspaceACL model correctly; it returns a GSON map
+   * instead. Run this through a typed Gson conversion process to coerce it into the desired type.
+   */
+  private static Map<String, WorkspaceAccessEntry> extractAclResponse(WorkspaceACL aclResp) {
+    Type accessEntryType = new TypeToken<Map<String, WorkspaceAccessEntry>>() {}.getType();
+    Gson gson = new Gson();
+    return gson.fromJson(gson.toJson(aclResp.getAcl(), accessEntryType), accessEntryType);
+  }
+
+  private static void backfill(
+      WorkspacesApi workspacesApi, String billingProjectPrefix, boolean dryRun)
+      throws ApiException {
+    int userUpgrades = 0;
+    int workspaceUpdates = 0;
+    for (WorkspaceResponse resp : workspacesApi.listWorkspaces()) {
+      Workspace w = resp.getWorkspace();
+      if (!w.getNamespace().startsWith(billingProjectPrefix)) {
+        continue;
+      }
+
+      String id = w.getNamespace() + "/" + w.getName();
+      if (!"PROJECT_OWNER".equals(resp.getAccessLevel())) {
+        log.warning(
+            String.format(
+                "service account has '%s' access to workspace '%s'; skipping",
+                resp.getAccessLevel(), id));
+        continue;
+      }
+
+      List<WorkspaceACLUpdate> updates = new ArrayList<>();
+      Map<String, WorkspaceAccessEntry> acl =
+          extractAclResponse(workspacesApi.getWorkspaceAcl(w.getNamespace(), w.getName()));
+      for (String user : acl.keySet()) {
+        WorkspaceAccessEntry entry = acl.get(user);
+        if (!CAN_COMPUTE_ROLES.contains(entry.getAccessLevel()) || entry.getCanCompute()) {
```

Comment:
I'd prefer reversing this check, along with a comment like "do not add canCompute if the user already has it or the user does not have a high enough level for upgrading"

Reply:
Sorry - I meant a smaller change than that, to:
`if (entry.getCanCompute() || !CAN_COMPUTE_ROLES.contains(entry.getAccessLevel()))`
but it's not important.  This is fine.

Reply:
Aha - sorry, I misunderstood. Done.